File: python/pn.py
#!/usr/bin/python

p = True
n = 100000
print("2\n3")

for i in range(5, n, 2):
    for j in range(3, int(i / 2) + 1, 2):
        if i % j == 0:
            p = False
            break
        else:
            p = True

    if p:
        print(i)
        p = False

File: perl/pn.pl
#!/usr/bin/perl

$p = 1;
$n = 100000;
print "2\n3\n";

for ($i = 5; $i < $n; $i += 2) {
    for ($j = 3; $j <= int($i / 2); $j += 2) {
        if ($i % $j == 0) {
            $p = 0;
            last;
        } else {
            $p = 1;
        }
    }

    if ($p == 1) {
        print $i . "\n";
        $p = 0;
    }
}
